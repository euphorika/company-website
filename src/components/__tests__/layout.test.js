import React from "react"
import { shallow } from "enzyme"
import toJson from 'enzyme-to-json'
import { useStaticQuery } from "gatsby"

import Layout from "../layout"

beforeEach(() => {
  useStaticQuery.mockReturnValueOnce({
    site: {
      siteMetadata: {
        title: `Default Starter`
      }
    }
  })
})

describe("Layout", () => {
  it("renders correctly", () => {
    const LayoutElement = (
      <Layout>
        <div>Children</div>
      </Layout>
    )

    const wrapper = shallow(LayoutElement)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
