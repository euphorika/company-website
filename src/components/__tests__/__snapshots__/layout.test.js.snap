// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
Array [
  <header
    className="siteHeader"
  >
    <div
      style={
        Object {
          "margin": "0 auto",
          "maxWidth": 960,
          "padding": "1.45rem 1.0875rem",
        }
      }
    >
      <h1
        style={
          Object {
            "margin": 0,
          }
        }
      >
        <a
          href="/"
          style={
            Object {
              "color": "white",
              "textDecoration": "none",
            }
          }
        >
          Default Starter
        </a>
      </h1>
    </div>
  </header>,
  <nav
    className="pageNavigation"
  >
    <ul>
      <li>
        <a
          href="#"
        >
          euphorika
        </a>
      </li>
      <li>
        <a
          href="#"
        >
          leistungen
        </a>
      </li>
      <li>
        <a
          href="#"
        >
          best cases
        </a>
      </li>
      <li>
        <a
          href="#"
        >
          jobs
        </a>
      </li>
      <li>
        <a
          href="#"
        >
          kontakt
        </a>
      </li>
    </ul>
  </nav>,
  <div
    className="innerContainer"
  >
    <main
      className="main"
    >
      <div>
        Children
      </div>
    </main>
  </div>,
  <footer
    class="pageFooter"
  >
    <ul>
      <li>
        <a
          href="/impressum/"
        >
          Impressum
        </a>
      </li>
      <li>
        <a
          href="/datenschutz/"
        >
          Datenschutz
        </a>
      </li>
    </ul>
  </footer>,
]
`;
